pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -742,21 +769,67 @@ func (l *LibvirtDomainManager) asyncMigrationAbort(vmi *v1.VirtualMachineInstanc
 	}(l, vmi)
 }
 
+func generateDomainName(vmi *v1.VirtualMachineInstance) string {
+	namespace := vmi.Namespace
+	name := vmi.Name
+	if vmi.Status.MigrationState != nil &&
+		vmi.Status.MigrationState.TargetState != nil &&
+		vmi.Status.MigrationState.TargetState.DomainName != nil &&
+		*vmi.Status.MigrationState.TargetState.DomainName != "" &&
+		vmi.Status.MigrationState.TargetState.DomainNamespace != nil &&
+		*vmi.Status.MigrationState.TargetState.DomainNamespace != "" {
+		name = *vmi.Status.MigrationState.TargetState.DomainName
+		namespace = *vmi.Status.MigrationState.TargetState.DomainNamespace
+	}
+	domainName := fmt.Sprintf("%s_%s", namespace, name)
```

Comment:
Don't we have to truncate this to the maximum domain name length of libvirt?

Reply:
I've seen it truncated before, but I guess libvirt takes care of that then. We just have to make sure it's properly handled in destxml as well

Reply:
I should probably use this function instead though https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/util/libvirt_helper.go#L408-L410

Reply:
I wish we'd switch to a static domain name... One day